### PR TITLE
Fixes #10183: Setting TTL to "30 minutes" in sharedfile to node generic method make it fail completly

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/10183

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

File: tree/30_generic_methods/sharedfile_to_node.cf [hunk 1]
```diff
@@ -95,6 +95,7 @@
 # @parameter file_id      unique name that will be used to identify the file on the receiver
 # @parameter file_path    path of the file to share
 # @parameter ttl          time to keep the file on the policy server in seconds or in human readable form (see long description)
+# @parameter_constraint ttl "regex" : "^(\\d+\\s*(days?|d))?(\\d+\\s*(hours?|h))?(\\d+\\s*(minutes?|m))?(\\d+\\s*(seconds?|s))?$"
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_to_node.cf]: 🎆

File: tree/30_generic_methods/sharedfile_to_node.cf [hunk 2]
```diff
@@ -57,7 +57,7 @@
 # * *minutes* or *m*
 # * *seconds* or *s*
 #
-# The ttl value can look like *1day 2hours 3minute 4seconds* or can be abbreviated in the form *5h 3s*.
+# The ttl value can look like *1 day 2hours 3m inute 4seconds* or can be abbreviated in the form *5h 3s*.
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_to_node.cf]: `3m` or `3 minutes` I guess.

**Comment by ncharles** [comment]
Commit modified

File: tree/30_generic_methods/sharedfile_to_node.cf (hunk 2 above)
Comment (ncharles) [inline tree/30_generic_methods/sharedfile_to_node.cf]: you are perfectly correct

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**